### PR TITLE
Suppress comment support

### PR DESCRIPTION
**Pull request by tronxd**
* Support of suppress a check for a definition block

### CODE REVIEW

File: checkov/terraform/checks/resource/base_check.py [hunk 1]
```diff
@@ -17,8 +17,11 @@ def __init__(self, name, id, categories, supported_resources):
         self.logger = logging.getLogger("{}".format(self.__module__))
         resource_registry.register(self)
 
-    def run(self, scanned_file, resource_configuration, resource_name, resource_type):
-        result = self.scan_resource_conf(resource_configuration)
+    def run(self, scanned_file, resource_configuration, resource_name, resource_type, suppress=False):
+        if suppress:
+            result = CheckResult.SUPPRESSED
```
Comment (schosterbarak) [inline checkov/terraform/checks/resource/base_check.py]: maybe "skipped" is a more matching name?

File: checkov/terraform/checks/resource/registry.py [hunk 2]
```diff
@@ -18,16 +18,20 @@ def get_checks(self, resource):
...
 
-    def scan(self, block, scanned_file):
+    def scan(self, block, scanned_file, skipped_checks):
         resource = list(block.keys())[0]
         resource_conf = block[resource]
         results = {}
-        for check in self.get_checks(resource):
+        checks = self.get_checks(resource)
+        for check in checks:
+            suppress = False
```
Comment (schosterbarak) [inline checkov/terraform/checks/resource/registry.py]: skipped?

File: checkov/terraform/context_parsers/base_parser.py [hunk 3]
```diff
@@ -32,16 +32,27 @@ def _filter_file_lines(self):
...
+    def _read_file_lines(self, tf_file):
         self.tf_file = tf_file
         with(open(tf_file, 'r')) as file:
             file.seek(0)
             file_lines = [(ind + 1, line) for (ind, line) in
                           list(enumerate(file.readlines()))]
-            self.file_lines = file_lines
+            return file_lines
+
+    def _collect_check_suppressions(self):
```
Comment (schosterbarak) [inline checkov/terraform/context_parsers/base_parser.py]: grammer is confusing on this method name

File: checkov/terraform/context_parsers/registry.py [hunk 4]
```diff
@@ -20,9 +20,8 @@ def enrich_definitions_context(self, definitions):
         for definition_type in definition_blocks_types.keys():
             if definition_type in supported_definitions:
                 context_parser = self.context_parsers[definition_type]
-                context_parser.read_file_lines(tf_file)
                 definition_blocks = definition_blocks_types[definition_type]
-                self.definitions_context[tf_file] = context_parser.enrich_definition_block(definition_blocks)
+                self.definitions_context[tf_file] = context_parser.run_context_parsing(tf_file,definition_blocks)
```
Comment (schosterbarak) [inline checkov/terraform/context_parsers/registry.py]: just "run" is enough as a method name

File: checkov/terraform/output/report.py [hunk 5]
```diff
@@ -98,15 +83,15 @@ def get_test_suites(self):
...
+            check_name = record.check_name
             if check_name not in test_cases:
                 test_cases[check_name] = []
 
-            test_name = "{} {}".format(check_name, record['resource'])
-            test_case = TestCase(name=test_name, file=record['file_path'], classname=record["check_class"])
-            if record['check_result'] == CheckResult.FAILURE:
+            test_name = "{} {}".format(check_name, record.resource)
+            test_case = TestCase(name=test_name, file=record.file_path, classname=record.check_class)
+            if record.check_result == CheckResult.FAILURE:
```
Comment (schosterbarak) [inline checkov/terraform/output/report.py]: we need to add the skipped tests to the junit xml

File: checkov/terraform/context_parsers/parsers/resource_context_parser.py [hunk 6]
```diff
@@ -23,9 +24,11 @@ def enrich_definition_block(self, block):
...
                 line_tokens = [x.replace('"', "") for x in line.split()]
                 if all(x in line_tokens for x in ['resource', resource_type, resource_name]):
-                    self.context[resource_type][resource_name]["start_line"] = line_num
-                    self.context[resource_type][resource_name]["end_line"] = self.compute_definition_end_line(line_num)
-
+                    start_line = line_num
+                    end_line = self._compute_definition_end_line(line_num)
+                    self.context[resource_type][resource_name]["start_line"] = start_line
+                    self.context[resource_type][resource_name]["end_line"] = end_line
+                    self.context[resource_type][resource_name]["code_lines"] = self.file_lines[start_line - 1: end_line]
```
[the remark this answers is not in the document]
Comment (tronxd) [reply checkov/terraform/context_parsers/parsers/resource_context_parser.py]: After we know what a parsing context should contain, we should refactor into an appropriate class

File: checkov/terraform/output/report.py [hunk 7]
```diff
@@ -45,10 +45,10 @@ def get_json(self):
...
         return {
             "results": {
-                "passed_checks": self.passed_checks,
-                "failed_checks": self.failed_checks,
-                "suppressed_checks": self.suppressed_checks,
-                "parsing_errors": self.parsing_errors
+                "passed_checks": [check.__dict__ for check in self.passed_checks],
+                "failed_checks": [check.__dict__ for check in self.failed_checks],
+                "suppressed_checks": [check.__dict__ for check in self.suppressed_checks],
+                "parsing_errors": [check for check in self.parsing_errors]
```
[the remark this answers is not in the document]
Comment (tronxd) [reply checkov/terraform/output/report.py]: no, it's a list of strings

**Pull request merged**